Makefile

```diff
@@ -32,5 +32,15 @@ site-docs: site
 	for f in site/manual/gh*.md; do sed -i.bak -e '/^### SEE ALSO/,$$d' "$$f"; done
 	rm -f site/manual/*.bak
 	git -C site add 'manual/gh*.md'
-	git -C site commit -m 'update help docs'
+	git -C site commit -m 'update help docs' || true
 .PHONY: site-docs
+
+site-publish: site-docs
+ifndef GITHUB_REF
+	$(error GITHUB_REF is not set)
+endif
+	sed -i.bak -E 's/(assign version = )".+"/\1"$(GITHUB_REF:refs/tags/v%=%)"/' site/index.html
```

Comment:
why specify .bak only to then delete the backup immediately? seems equivalent to just omitting .bak

Reply:
It's a good question! The reason is that GNU sed and BSD sed differ on how they parse and handle `-i` with empty extension (edit file in place). To avoid differences between running something on Linux vs. running it on a system like macOS, I prefer to always specify a backup extension and then manually remove it.

Reply:
ah, interesting. i guess i never used BSD sed enough to trip on that.